Update Fast Refresh doc (#15923)
vercel/next.js: docs/basic-features/fast-refresh.md

Sometimes, this can lead to unexpected results. For example, even a `useEffect`
with an empty array of dependencies would still re-run once during Fast Refresh.

> However, writing code resilient to occasional re-running of useEffect is a good practice even without Fash Refresh. It will make it easier for you to introduce new dependencies to it later on and it's enforced by [React Strict Mode](/docs/api-reference/next.config.js/react-strict-mode), which we highly recommend enabling.

Reviewer: My suggestion is still different to what you changed, I added a new line and used multiple lines for the paragraph, due to this section being a bit different in the way it's written compared to others.

```suggestion

However, writing code resilient to occasional re-running of useEffect is a good practice even
without Fash Refresh. It will make it easier for you to introduce new dependencies to it later on 
and it's enforced by [React Strict Mode](/docs/api-reference/next.config.js/react-strict-mode),
which we highly recommend enabling.
```

> However, writing code resilient to occasional re-running of useEffect is a good practice because it not only ensures you always write resilient code but also because it's enforced by React Strict Mode. Additionally, this makes it easier for you to introduce new dependencies to it later on. To learn more about React Strict Mode refer to the React Strict Mode docs https://reactjs.org/docs/strict-mode.html.

Reviewer: 😎 👍 

```suggestion
However, writing code resilient to occasional re-running of useEffect is a good practice even without Fash Refresh, as it's also enforced by [React Strict Mode](/docs/api-reference/next.config.js/react-strict-mode). Additionally, this makes it easier for you to introduce new dependencies to it later on.
```
